bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/ComponentImplTest.java

```diff
@@ -61,13 +79,32 @@ public void testReferencedComponentId() {
         Component component = getReferencedComponentUnderTest(XF_COMPONENT_1, TEST_PAGE_EN, XF_COMPONENT_10);
         Assertions.assertEquals("experiencefragment-789e0d5de3", component.getId(), "ID mismatch");
     }
-
+    
+    
+    @Test
+    public void testStyleSystemClasses()
+    {
+    	final String WE_RETAIL_TITLE = TEST_PAGE_EN+"/jcr:content/root/title_core";
+    	Object[] mappings = null;
+    	String[] styleClasses = getComponentUnderTest(WE_RETAIL_TITLE,mappings).getAppliedCssClasses().split(StringUtils.SPACE);
+    	assertTrue(styleClasses!=null && styleClasses.length == 2);
```

Comment:
Potential NPE, `styleClasses` can't possibly be null because `String#split()` doesn't return null. 
However, `Component#getAppliedCssClasses()` does potentially return null.
The null check should be done with `Assertions#assertNotNull()`.

Check of the array value should be done with `Assertions.assertArrayEquals()` instead of just checking the length;; or, since `getAppliedCssClasses()` returns a string instead of an array, just check the string value against the expected value. 

Additionally, should check the other situations, i.e. when the string is empty and/or not set.

Reply:
@ky940819 - Please take a look at the updated code. Since we are just calling another re-usable function, I was testing for just 1 usecase, so we dont repeat re-testing all the use cases that were originally tested in ComponentStyleInfo.class. However, if you feel it would be nice to do the due deligence. Please let me know. I originally had those test cases before we brought this ComponentStyleInfo reference. Thanks in advance !

Reply:
I was just thinking that you have logic that returns null (i.e. when string is blank, or is null). 
It would be nice to have this covered such that if It is ever changed to return something else in that case, it would be caught by the tests. 

I'm not suggesting re-testing all of ComponentStyleInfo, I'm just thinking of the situations:

- `Mockito.doReturn("   ").when(componentStyleInfoMock).getAppliedCssClasses();` -> null
- `Mockito.doReturn(null).when(componentStyleInfoMock).getAppliedCssClasses();` -> null

I'd also consider changing
```JAVA
String[] styleClasses = getComponentUnderTest(WE_RETAIL_TITLE,mappings).getAppliedCssClasses().split(StringUtils.SPACE);
assertTrue(styleClasses!=null && styleClasses.length == 2);
```

to
```JAVA
String[] styleClasses = getComponentUnderTest(WE_RETAIL_TITLE,mappings).getAppliedCssClasses();
Assertions.assertNotNull(styleClasses);
Assertions.assertEquals("class1 class2", styleClasses);
```

Reply:
@ky940819 - This is now taken care of !